core/modules/story.js

```diff
@@ -97,7 +97,8 @@ Story.prototype.addToHistory = function(navigateTo,navigateFromClientRect) {
 	$tw.utils.each(titles,function(title) {
 		historyList.push({title: title, fromPageRect: navigateFromClientRect});
 	});
-	this.wiki.setTiddlerData(this.historyTitle,historyList,{"current-tiddler": titles[titles.length-1]});
+	var current = titles[titles.length-1];
+	this.wiki.setTiddlerData(this.historyTitle,historyList,{"current-tiddler": current, "focused-tiddler": current});
```

Comment:
The 'focused-tiddler' stuff appears to be mostly but not entirely within a plugin, why is that?

Reply:
I'm using the built in core mechanism, that is triggered by $:/HistoryList changes only. There is no new code.

The new focus plugin only catches changes if you scroll. 

The problem is: If you click a link in the right sidebar, it triggers a "scroll-into-view". Which is triggered by changing the $:/HistoryList in navigator.js. navigator catches link events. 

The `module-type: storyview` modules like classic.js. are responsible for the animation. After animation is done there is no other signal, to highlight the "now focused" tiddler.

The new focus-highlighting in toc.tid is done with a filter like this:

```
[all[current]]-[<tv-history-list>get[focused-tiddler]]
```

The $:/HistoryList is updated in navigator.js with `this.addToHistory(event.navigateTo,event.navigateFromClientRect);`

which then calls a wiki.js function: `this.wiki.addToHistory(title,fromPageRect,this.historyTitle);` 

which is implemented like this: 

```
exports.addToHistory = function(title,fromPageRect,historyTitle) {
	var story = new $tw.Story({wiki: this, historyTitle: historyTitle});
	story.addToHistory(title,fromPageRect);
};
```

So story.js is the "lowest" module which does the job. -> That's why I needed it to update `focused-tiddler`. Which now can be used also without the plugin in a backwards compatible way. 

Which could be seen in the video :/

Reply:
I think, calling `this.wiki.addToHistory` in navigator.js is not the right thing to do. ... But it's the only way do get the work done without changing more elements of the core.